src/tool/subcommands/car_cmd.rs

```diff
@@ -57,15 +56,15 @@ impl CarCommands {
                     .try_collect()
                     .await?;
 
-                let all_roots = NonEmpty::from_vec(
+                let all_roots = NonEmpty::new(
                     car_streams
                         .iter()
                         .flat_map(|it| it.header.roots.iter())
                         .unique()
                         .cloned()
                         .collect_vec(),
```

Comment:
There seems to be an opportunity to save the `collect_vec` call if `NonEmpty` can be constructed from an iterator

Reply:
It can't be contracted from an iterator - the iterator might be empty